Update `booleans` concept (#3131)
exercism/go: concepts/booleans/about.md

> # About

Reviewer: Looks good! Just needs syncing across the `md` files.

A Boolean value represents whether a condition is `true` or `false`.

> In Go, the `bool` type has only those two values.

Reviewer: ```suggestion
In Go, the `bool` type can only be one of those two values.
```

```go
isDoorLocked := true
hasKey := true
knowsCode := false
```

## Logical Operators

> Logical operators work on one or two `bool` values and produce a Boolean result.

Reviewer: ```suggestion
Logical operators work on one or two Boolean values and produce a Boolean result.
```

| Operator | Name | Behavior |
| -------- | ---- | -------- |
| `!`      | NOT  | Returns the opposite of a single Boolean value |
| `&&`     | AND  | Returns `true` only when both sides are `true` |

> | `||`     | OR   | Returns `true` when at least one side is `true` |

Reviewer: ```suggestion
| `\|\|`     | OR   | Returns `true` if any side is `true` |
```

Use `!` (NOT) when you need the opposite of a Boolean value:

```go
!isDoorLocked // false
```

Use `&&` (AND) when both conditions must be `true`:

```go
canUnlockDoor := isDoorLocked && hasKey // true
```

Use `||` (OR) when at least one condition must be `true`:

```go
canOpenDoor := canUnlockDoor || knowsCode // true
```

Boolean operators are evaluated by default in this order: `!` first, then `&&`, then `||`.

> Use parentheses to override this behavior:

Reviewer: ```suggestion
Boolean operators are evaluated in this order: first `!`, then `&&`, then `||`.
Use parentheses to change the order of evaluation:
```

"By default" sounds like you can change the precendence.

Author: The second sentence is a bit tricky, and I'll have to think about it. The parentheses are being used to group expressions, and changing how Go would otherwise group expressions affects the evaluation order. `Use parentheses to change which operations are evaluated first:` might work, but then we should highlight the order with and without parentheses in the code example.
